forge/routes/ui/index.js

```diff
@@ -69,6 +69,10 @@ module.exports = async function (app) {
                 injection += `<script>window.sentryConfig = { dsn: "${telemetry.frontend.sentry.dsn}", production_mode: ${telemetry.frontend.sentry.production_mode ?? true}, version: "flowfuse@${config.version}", environment: "${process.env.SENTRY_ENV ?? (process.env.NODE_ENV ?? 'unknown')}" }</script>`
             }
 
+            if (config.base_url) {
+                injection += `<link rel="canonical" href="${config.base_url}" />`
```

Comment:
@hardillb @knolleary This doesn't work as you're not including the path element. E.g. `/accounts/create` is now just pointing to the base URL?

Than this seems to be included an _all_ pages, so if any page later will be more open to be more like a github for NR, all cacnonical URLs will be wrong.

Reply:
yes, this applies to all pages.

We can revert it for now and look to see if we can apply the route as well as the host.

Reply:
I don't think the revert is worth it, though a follow up should be scheduled. Not sure if VueJS for example can update the link tag dynamically?

Reply:
This might be useful

https://www.digitalocean.com/community/tutorials/vuejs-vue-seo-tips

Reply:
@ZJvandeWeg https://github.com/FlowFuse/flowfuse/issues/3298

Reply:
@ZJvandeWeg @hardillb A crawler shouldn't really be indexing the pages _after_ login anyway, so setting the canonical URL to the login page should be good enough!

Reply:
@Pezmc There's still two different pages, one needed a path set, the other just worked by the base_url as there's no path. So I'm not quite sure what you're saying with your comment.

Reply:
The canonical tag is only used by search engines to determine the “main” version of the page. The only pages that the search engines can access are the public ones, mainly login and register (I'm not currently aware of any other pages that are do not require authentication, search engines/crawlers are not able to bypass auth).

Having the canonical tag on pages behind auth is somewhat redundant so we can focus on the tag having the correct values on these two.

As @hardillb mentioned, there are ways to dynamically alter meta tags but we should focus to add them only where they matter. 

I'd go with hardcoded values if the pages mentioned above are the only publicly accessible ones, there's no need for the overhead of an additional library IMO.